vault/auth.go

```diff
@@ -560,6 +549,26 @@ func (c *Core) newCredentialBackend(
 		return nil, fmt.Errorf("unknown backend type: %s", t)
 	}
 
+	// Set up conf to pass in plugin_name
+	conf := make(map[string]string, len(entry.Options)+1)
+	for k, v := range entry.Options {
+		conf[k] = v
+	}
+	if entry.Config.PluginName != "" {
+		conf["plugin_name"] = entry.Config.PluginName
+	}
+
+	uuidBytes, err := uuid.ParseUUID(entry.UUID)
+	if err != nil {
+		return nil, err
+	}
+	sum := sha256.Sum256(uuidBytes)
+	deterministicID, err := uuid.FormatUUID(sum[0:16])
+	if err != nil {
+		return nil, err
+	}
+	conf["uid"] = deterministicID
```

Comment:
Instead of deriving this from the mount UUID, would it make sense to provide a unique "seed" value with every mount and just store it with the mount entry.  This value would have no connection with the storage and would just be a unique mapping created at mount time.  This wouldn't be too dissimilar to when we added mount accessor.

Reply:
heh, already was working on this. See the latest commits